packages/components/src/internal/components/user/APIKeysPanel.tsx

```diff
@@ -111,30 +124,55 @@ export const KeyGenerator: FC<KeyGeneratorProps> = props => {
         document.execCommand('copy');
     }, [keyValue]);
 
+    const changeDescription = useCallback((event: ChangeEvent<HTMLInputElement>) => {
+        setDescription(event.target.value);
+    }, []);
+
     return (
-        <>
-            <div className="top-spacing form-group">
-                <button className="btn btn-success api-key__button" onClick={onGenerateKey} disabled={!!keyValue}>
-                    Generate {noun}
-                </button>
-
-                <input
-                    disabled
-                    type="text"
-                    className="form-control api-key__input"
-                    name={type + '_token'}
-                    value={keyValue}
-                />
-                <button
-                    className="btn btn-default api-key__button"
-                    title="Copy to clipboard"
-                    name={'copy_' + type + '_token'}
-                    onClick={onCopyKey}
-                    disabled={!keyValue}
-                >
-                    <i className="fa fa-clipboard"></i>
-                </button>
-            </div>
+        <Modal
+            title={noun}
+            cancelText={keyValue ? 'Done' : undefined}
+            onCancel={onClose}
+            canConfirm={!keyValue}
+            confirmText={type === 'apikey' ? 'Generate ' + noun : undefined}
+            onConfirm={!keyValue && type === 'apikey' ? onGenerateKey : undefined}
+        >
+            {type === 'apikey' && !keyValue && (
+                <div>
+                    <label htmlFor="keyDescription" className="right-spacing">
+                        Description (optional)
+                    </label>
+                    <input
+                        className="form-control api-key__input"
+                        id="keyDescription"
+                        type="text"
+                        onChange={changeDescription}
+                        autoFocus
+                        disabled={!!keyValue}
+                    />
+                </div>
+            )}
+            {keyValue && <div className="api-key__description">{description}</div>}
+            {!!keyValue && (
```

Comment:
minor: is there a reason that one of these checks `!!keyValue` and the other doesn't? Seems like they could/should match.

Reply:
No good reason. I joined them all under one test for `!!keyValue`